protocol/packages/dex/src/impl_/out_local.rs

```diff
@@ -769,13 +769,18 @@ mod impl_contract {
         for State<SwapTask, SwapGroup, SwapClient, ForwardToInnerMsg, ForwardToInnerContinueMsg>
     where
         SwapTask: SwapTaskT
-            + ContractInSwap<TransferOutState, <SwapTask as SwapTaskT>::StateResponse>
-            + ContractInSwap<SwapState, <SwapTask as SwapTaskT>::StateResponse>
-            + ContractInSwap<TransferInInitState, <SwapTask as SwapTaskT>::StateResponse>
-            + ContractInSwap<TransferInFinishState, <SwapTask as SwapTaskT>::StateResponse>,
+            + ContractInSwap<TransferOutState, StateResponse = <SwapTask as SwapTaskT>::StateResponse>
+            + ContractInSwap<SwapState, StateResponse = <SwapTask as SwapTaskT>::StateResponse>
+            + ContractInSwap<
+                TransferInInitState,
+                StateResponse = <SwapTask as SwapTaskT>::StateResponse,
+            > + ContractInSwap<
+                TransferInFinishState,
+                StateResponse = <SwapTask as SwapTaskT>::StateResponse,
+            >,
         ForwardToInnerMsg: ForwardToInner,
     {
-        type StateResponse = SwapTask::StateResponse;
+        type StateResponse = <SwapTask as SwapTaskT>::StateResponse;
```

Comment:
Because it becomes ambiguous when the generic parameter becomes an associated type. That can be "mitigated"/worked around if the two associated types have different names.